File: DONGUK/프로그래머스_WEEK 6/카펫.py
def solution(brown, yellow):
    answer = []
    cnt = brown + yellow
    
    candidate = []
    for i in range(2, cnt):
        if cnt % i == 0:
            q = cnt // i
            r = cnt // q
            
            if q >= r:
                candidate.append((q, r))
    
    for data in candidate:
        garo, sero = data[0], data[1]
        
        sum_ = (garo * 2) + (sero - 2) * 2
        
        if sum_ == brown:
            answer = data
            break
    return answer

File: KIJUNG/PRO_WEEK_06/카펫.cpp
#include <vector>
#include <math.h>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    int sum = brown+yellow;
    for(int i=1; i<=sqrt(yellow); i++){
        if(yellow%i == 0){
            if((i+2)*(yellow/i+2) == sum){
                answer.push_back(yellow/i+2);
                answer.push_back(i+2);
            }
        }
    }
    return answer;
}